cypress/e2e/sliding-sync/sliding-sync.ts

```diff
@@ -0,0 +1,333 @@
+import { SettingLevel } from "../../../src/settings/SettingLevel";
+import { Layout } from "../../../src/settings/enums/Layout";
+import { ProxyInstance } from "../../plugins/sliding-sync";
+
+describe("Sliding Sync", () => {
+    beforeEach(() => {
+        cy.startSynapse("default").as("synapse").then(synapse => {
+            cy.startProxy(synapse).as("proxy");
+        });
+
+        cy.all([
+            cy.get<SynapseInstance>("@synapse"),
+            cy.get<ProxyInstance>("@proxy"),
+        ]).then(([synapse, proxy]) => {
+            cy.enableLabsFeature("feature_sliding_sync");
+
+            cy.intercept("/config.json?cachebuster=*", req => {
+                return req.continue(res => {
+                    res.send(200, {
+                        ...res.body,
+                        setting_defaults: {
+                            feature_sliding_sync_proxy_url: `http://localhost:${proxy.port}`,
+                        },
+                    });
+                });
+            });
+
+            cy.initTestUser(synapse, "Sloth").then(() => {
+                return cy.window({ log: false }).then(() => {
+                    cy.createRoom({ name: "Test Room" }).as("roomId");
+                });
+            });
+        });
+    });
+
+    afterEach(() => {
+        cy.get<SynapseInstance>("@synapse").then(cy.stopSynapse);
+        cy.get<ProxyInstance>("@proxy").then(cy.stopProxy);
+    });
+
+    // assert order
+    const checkOrder = (wantOrder: string[]) => {
+        cy.contains(".mx_RoomSublist", "Rooms").find(".mx_RoomTile_title").should((elements) => {
+            expect(_.map(elements, (e) => {
+                return e.textContent;
+            }), "rooms are sorted").to.deep.equal(wantOrder);
+        });
+    };
+    const bumpRoom = (alias: string) => {
+        // Send a message into the given room, this should bump the room to the top
+        cy.get<string>(alias).then((roomId) => {
+            return cy.sendEvent(roomId, null, "m.room.message", {
+                body: "Hello world",
+                msgtype: "m.text",
+            });
+        });
+    };
+    const createAndJoinBob = () => {
+        // create a Bob user
+        let baseUrl;
+        let creds;
+        cy.get<SynapseInstance>("@synapse").then((synapse) => {
+            baseUrl = synapse.baseUrl;
+            return cy.registerUser(synapse, "bob", "passwordgoeshere123", "Bob");
+        }).then((c) => {
+            creds = c;
+            return cy.window({ log: false });
+        }).then((window) => {
+            return window.matrixcs.createClient({
+                baseUrl: baseUrl,
+                accessToken: creds.accessToken,
+                userId: creds.userId,
+                deviceId: creds.deviceId,
+            });
+        }).as("bob");
+
+        // invite Bob to Test Room and accept then send a message.
+        cy.all([cy.get<string>("@roomId"), cy.get<MatrixClient>("@bob")]).then(([roomId, bob]) => {
```

Comment:
If the bot should always join the rooms he is invited to, `getBot` supports the `autoAcceptInvites` option.

Reply:
Empirically, this is racey though as the test doesn't know when the join is complete, resulting in flakey tests failing because the bot isn't in the room.